types/src/auction/providers.rs

```diff
@@ -9,13 +9,10 @@ use crate::{
 /// Provider of runtime host functionality.
 pub trait RuntimeProvider {
     /// This method should return the caller of the current context.
-    fn get_caller(&self) -> AccountHash;
+    fn get_caller(&self) -> Result<AccountHash, Error>;
 
     /// Gets named key under a `name`.
-    fn get_key(&self, name: &str) -> Option<Key>;
-
-    /// Puts key under a `name`.
-    fn put_key(&mut self, name: &str, key: Key);
+    fn get_key(&self, name: &str) -> Result<Option<Key>, Error>;
```

Comment:
Why does this need to be a `Result`?

Reply:
https://github.com/CasperLabs/casper-node/pull/684#discussion_r547212160

Reply:
(see above)